Lib/test/test_format.py

```diff
@@ -304,9 +304,9 @@ def test_str_format(self):
         test_exc('%c', sys.maxunicode+1, OverflowError,
                  "%c arg not in range(0x110000)")
         #test_exc('%c', 2**128, OverflowError, "%c arg not in range(0x110000)")
-        test_exc('%c', 3.14, TypeError, "%c requires int or char")
-        test_exc('%c', 'ab', TypeError, "%c requires int or char")
-        test_exc('%c', b'x', TypeError, "%c requires int or char")
+        test_exc('%c', 3.14, TypeError, "%c requires an int or a unicode character, not float")
```

Comment:
Looks like this `test_exc` helper function has a bug: it doesn't re-raise unexpected exceptions, which is why the test is still passing. Could you add `raise` around line 84 to fix this?

Reply:
Done.  As it was mentioned in the issue, I'm not sure if this is not intentional.

Reply:
Thanks. Pretty sure it was simply an oversight, though I haven't delved into the git history.